### DISCUSSION TITLE
what can be the root cause of non determinism result?

### DISCUSSION THREAD

**Discussion by 0x8f701** [gfx-rs/wgpu#6118] [Q&A] [answered]
I observed a weird the thing:

- the same shaderr
- the same input
- the same macos-latest CI enviroment (m1)

it can produce different result time to time, on Linux it always output the same result but on macos m1 it fails time to time

what can be the root cause? I have no clue on how to fix it

**Comment by 0x8f701** [comment]
also sometimes it even fails to create compute pipeline with validation error. If I rerun the CI it succeeds magically. I'm using wgpu@22.1.0 with wgsl, webgpu features enabled

**Comment by ErichDonGubler** [comment]
Unfortunately, there's not enough information in your OP to answer this question. The best way to help us out is to provide a small reproducible example. It's also extremely helpful to include the validation error message, to ensure others are reproducing the same issue.

**Comment by 0x8f701** [comment]
reproduce repro:

https://github.com/GopherJ/webgpu-shaders

what I'm suffering is that I got different result time to time, I have the same code in another repo and the CI sometimes succeeeds sometimes not

**Reply by 0x8f701** [reply]
You can test directly via:

```
cargo test --release --no-default-features --features std -- --test-threads 1 --nocapture
```

Or test in browser(you'll need to enbable webgpu & vulkan):

```
wasm-pack test --release --chrome  --features no_std
```

wasm test failed and the cargo test fails time to time

**Comment by 0x8f701** [comment]
this should be related to https://github.com/gfx-rs/wgpu/issues/6124

**Comment by 0x8f701** [comment, marked as answer]
fixed: https://github.com/gfx-rs/wgpu/issues/6124#issuecomment-2295115055
